sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -48,12 +48,27 @@ public BinaryData EventBody
             set => _amqpMessage.Body = AmqpMessageBody.FromData(MessageBody.FromReadOnlyMemorySegment(value.ToMemory()));
         }
 
-        BinaryData IMessageWithContentType.Data
+        BinaryData IMessageWithMetadata.Data
         {
             get => EventBody;
             set => EventBody = value;
         }
 
+        IDictionary<string, object> IMessageWithMetadata.Metadata
+        {
+            get
+            {
+                _metadata ??= new AnnotatedMessageMetadataDictionary(GetRawAmqpMessage())
```

Comment:
Why do we need to explicitly add the `ContentType` member?   The dictionary has access to the AMQP message;  can't it build that itself internally?